sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_custom_form_client.py

```diff
@@ -0,0 +1,296 @@
+            if not extracted_form.analyze_result.document_results:
+                raise HttpResponseError("Cannot call begin_extract_labeled_forms() with the ID of a model trained "
+                                        "without labels. Please call begin_extract_forms() instead.")
+            form_result = prepare_labeled_result(extracted_form, include_text_details)
+            return form_result
+
+        return self._client.begin_analyze_with_custom_model(
+            file_stream=form,
+            model_id=model_id,
+            include_text_details=include_text_details,
+            content_type=content_type,
+            cls=callback,
+            polling=LROBasePolling(timeout=POLLING_INTERVAL, **kwargs),
+            **kwargs
+        )
+
+    @distributed_trace
+    def delete_custom_model(self, model_id, **kwargs):
+        # type: (str, Any) -> None
+        """Mark model for deletion. Model artifacts will be permanently removed within a predetermined period.
+
+        Delete Custom Model.
+
+        :param model_id: Model identifier.
+        :type model_id: str
+        :rtype: None
+        :raises: ~azure.core.exceptions.HttpResponseError
+        """
+        self._client.delete_custom_model(
+            model_id=model_id,
+            **kwargs
+        )
+
+    @distributed_trace
+    def list_custom_models(self, **kwargs):
+        # type: (Any) -> Iterable[ModelInfo]
+        """List Custom Models.
+
+        :return: ItemPaged[~azure.ai.formrecognizer.ModelInfo]
+        :raises: ~azure.core.exceptions.HttpResponseError
+        """
+        return self._client.list_custom_models(
+            cls=lambda objs: [ModelInfo._from_generated(x) for x in objs],
+            **kwargs
+        )
+
+    @distributed_trace
+    def get_models_summary(self, **kwargs):
+        # type: (Any) -> ModelsSummary
+        """Get information about all custom models.
+
+        :return: Summary of models on account - count, limit, last updated.
+        :rtype: ~azure.ai.formrecognizer.ModelsSummary
+        :raises: ~azure.core.exceptions.HttpResponseError
+        """
+        response = self._client.get_custom_models(**kwargs)
+        return ModelsSummary._from_generated(response.summary)
+
+    @distributed_trace
+    def get_custom_model(self, model_id, **kwargs):
+        # type: (str, Any) -> CustomModel
+        """Get detailed information about a custom model.
+
+        :param str model_id: Model identifier.
+        :return: CustomModel
+        :rtype: ~azure.ai.formrecognizer.CustomModel
+        :raises: ~azure.core.exceptions.HttpResponseError
+        """
+        response = self._client.get_custom_model(model_id=model_id, include_keys=True, **kwargs)
+        if response.keys:
+            return CustomModel._from_generated(response)
+        raise HttpResponseError(message="Model id '{}' is a model that was trained with labels. "
```

Comment:
Do we want to pass the response too?

Reply:
This is some weirdness right now where response is actually a deserialized model so I can't pass it into `HttpResponseError`. 

We are separating supervised/unsupervised training/extracting for this preview, but hopefully in later previews when this is combined I won't have to do this check

Reply:
This should probably be a ValueError based on how it behaves now.

Reply:
Good point, changing to `ValueError`